luigi/contrib/kubernetes.py

```diff
@@ -63,19 +64,20 @@ class kubernetes(luigi.Config):
 class KubernetesJobTask(luigi.Task):
 
     __POLL_TIME = 5  # see __track_job
-    kubernetes_config = kubernetes()
+    kubernetes_config = Kubernetes()
 
     def _init_kubernetes(self):
         self.__logger = logger
         self.__logger.debug("Kubernetes auth method: " + self.auth_method)
-        if(self.auth_method == "kubeconfig"):
+        if self.auth_method == "kubeconfig":
             self.__kube_api = HTTPClient(KubeConfig.from_file(self.kubeconfig_path))
-        elif(self.auth_method == "service-account"):
+        elif self.auth_method == "service-account":
             self.__kube_api = HTTPClient(KubeConfig.from_service_account())
         else:
             raise ValueError("Illegal auth_method")
         self.job_uuid = str(uuid.uuid4().hex)
-        self.uu_name = self.name + "-luigi-" + self.job_uuid
+        now = datetime.utcnow()
+        self.uu_name = "%s-%s-%s" % (self.name, now.strftime('%Y%m%d%H%M%S'), self.job_uuid[:16])
```

Comment:
I like to have the date in the job name, but lsn't it nice to have `luigi` in it as well? Any argument against it? Also why are you taking only 16 chars from the hex ID?

Reply:
Right, so the reason is that Kubernetes constrains job names to be <64 characters. I ran into issues when I added the date, so I tried to reduce the amount of characters that are added to the name. 
After removing `-luigi` and halving the uuid, there are 32 characters available for the name (otherwise there would be 9 characters max for the jobname).

Reply:
Sounds good.